__tests__/endpoints.test.js

```diff
@@ -46,6 +47,40 @@ describe("/api", () => {
   });
 });
 
-describe("/api/articles/1", () => {
-    
-})
\ No newline at end of file
+4-articles-by-id
+describe("api/articles/aritcle_id", () => {
+  test("GET 200, responds with a sngular article object", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        const { article } = body;
+        expect(article.article_id).toBe(1);
+        expect(typeof article.title).toBe("string");
+        expect(typeof article.author).toBe("string");
+        expect(typeof article.body).toBe("string");
+        expect(typeof article.created_at).toBe("string");
+        expect(typeof article.votes).toBe("number");
+        expect(typeof article.topic).toBe("string");
+        expect(typeof article.article_img_url).toBe("string");
+      });
+  });
+  test("GET 404: sends an error message when given a valid but non-exists id", () => {
+    return request(app)
+      .get("/api/articles/999")
+      .expect(404)
+      .then(({ body} ) => {
+        const { message } = body;
+        expect(message).toBe("invalid query");
+      });
+  });
+  test.only("GET 400: sends an error message when given an invalid id", () => {
```

Comment:
we need to delete .only off below pushing - otherwise we could be letting lets of errors be pushed up